access reviews to v1 - using doc stub generator (#12207)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/resources/principalresourcemembershipsscope.md

---
title: "principalResourceMembershipsScope resource type"
description: "Allows for the selection of access review scope to review access of the selected principals to the selected resources."
author: "isabelleatmsft"
localization_priority: Normal
ms.prod: "governance"
doc_type: resourcePageType
---

# principalResourceMembershipsScope resource type

Namespace: microsoft.graph

> The principalResourceMembershipsScope is a type of [accessReviewScope](accessreviewscope.md) which allows you to select a collection of principal scopes and a collection of resource scopes and review access of selected principals to selected resources. See the supported queries to see what can be selected. It is used as the `scope` property of an [accessReviewScheduleDefinition](accessreviewscheduledefinition.md).

Reviewer: ```suggestion
The principalResourceMembershipsScope is a type of [accessReviewScope](accessreviewscope.md) that allows you to select a collection of principal scopes and a collection of resource scopes and review access of selected principals to selected resources. See the supported queries to see what can be selected. It is used as the `scope` property of an [accessReviewScheduleDefinition](accessreviewscheduledefinition.md).
```

Inherits from [accessReviewScope](../resources/accessreviewscope.md).

## Properties
|Property|Type|Description|
|:---|:---|:---|

> |principalScopes|[accessReviewScope](../resources/accessreviewscope.md) collection|Defines the scopes of the principals whose access to resources will be reviewed in the access review.|

Reviewer: ```suggestion
|principalScopes|[accessReviewScope](../resources/accessreviewscope.md) collection|Defines the scopes of the principals whose access to resources is reviewed in the access review.|
```

> |resourceScopes|[accessReviewScope](../resources/accessreviewscope.md) collection|Defines the scopes of the resources for which access will be reviewed to.|

Reviewer: ```suggestion
|resourceScopes|[accessReviewScope](../resources/accessreviewscope.md) collection|Defines the scopes of the resources for which access is reviewed.|
```

## Relationships
None.

### Supported queries for resourceScope

> The queries are supported as the `resourceScope` property. They determine the set of resources access is being reviewed to. 

Reviewer: ```suggestion
The queries are supported as the `resourceScope` property and they determine the set of resources for which access is being reviewed. 
```

|Scenario| resourceScope Query | 
|--|--|
| Reviewing access of principalScopes to a service principal | /servicePrincipals/{service principal ID} |
| Reviewing access of principalScopes to an Azure AD directory role | /roleManagement/directory/roleDefinitions/{role ID} |
| Reviewing access of principalScopes to all Azure AD directory roles | /roleManagement/directory/roleDefinitions |

### Supported queries for principalScope

> The queries are supported as the `principalScope` property. They determine the set of principals whose access to the associated resourceScope will be reviewed. The associated principalScope Type lists the odata query types accepted as the principalScope.

Reviewer: ```suggestion
The queries are supported as the `principalScope` property. They determine the set of principals whose access to the associated resourceScope is reviewed. The associated principalScope lists the OData query types accepted as the principalScope.
```
